client.go

```diff
@@ -256,6 +256,16 @@ func (c *Client) Start() {
 			}
 
 			time.Sleep(dur)
+
+			// exit if closed
+			c.mu.Lock()
```

Comment:
+ while you at it please also create `closed() bol` / `setClosed(b bool)` and hide mutexes there